src/java/com/puppetlabs/puppetserver/ShellUtils.java

```diff
@@ -72,15 +73,18 @@ private static ExecutionResult executeExecutor(CommandLine commandLine,
         //    PipedInputStream stdoutInputStream = new PipedInputStream(stdoutOutputStream);
         // but this requires that the input stream be read on a different thread
         // than this one. this is currently out of scope.
-        ByteArrayOutputStream stdoutOutputStream;
+        ByteArrayOutputStream stdoutOutputStream = new ByteArrayOutputStream();
```

Comment:
Yeah, we could still do something like the `PipedOutputStream` rather than writing to a `ByteArrayOutputStream` and having to read the bytes back from that.  Might save a bit of memory if we did it that way instead.  I don't think my latest changes helps or hurts with this.